Add Behavior GUI application

blourenco217 opened the pull request:
# Add Initial Behavior GUI Application

This pull request introduces the first version of the Behavior GUI application to the repository. The App allows the configuration of the device. It follows the new applications already released for the Olfactometer and SyringePump devices.
All functionalities present in the prior LabVIEW application are still accessible here, with added features such as the monitoring of DI3.

## Features
   - Device configuration.
   - Real-time monitoring of these configurations.
   - Built with .NET LTS (8.0) and Avalonia version (11.2.6).
   - Cross-platform UI with light/dark theme support.
   - Built upon the latest (up-to date) Interface.

blourenco217 committed:
First commit of the app

blourenco217 committed:
Testing different layouts

blourenco217 committed:
Digital Outputs Start and Stop PWM buttons added

blourenco217 committed:
Toggle buttons at the PWMs implemented

blourenco217 committed:
New tab item tested out

blourenco217 committed:
Display Inputs

blourenco217 committed:
new configuration

blourenco217 committed:
Added Apply buttons for RGBs LEDs and DOs

blourenco217 committed:
Redesign of analog input

blourenco217 committed:
Added events tab

blourenco217 committed:
Deleted extra commands in operations

blourenco217 committed:
Apply specific buttons to the extended features

blourenco217 committed:
Added input/output direction combo box for the DIOs

blourenco217 committed:
Tested out teh black boxes

blourenco217 committed:
PORT 1 stack completed with black boxes

blourenco217 committed:
Grid completed

blourenco217 committed:
Added a review view of the DIs and corrected some typos

blourenco217 committed:
Added apply button at the timestamp

blourenco217 committed:
Set and Clear Buttons were made interactive @ DIOs and @ RGBs and LEDs

blourenco217 committed:
Added same configuration and table view to every tab, added button apply specific to DIOs pulse configuration

blourenco217 committed:
Added remining Apply buttons

blourenco217 committed:
Improved view for the mimic control cells

blourenco217 committed:
RGBs and LEDs: made the set and clear interactive; corrected apply buttons; added button for the current control and the color control

blourenco217 committed:
Advanced Controls: same width for the 1st column

blourenco217 committed:
Made the start and stop buttons intetactive as well as then enable and disable

blourenco217 committed:
Reviewed servo motors outlook at the remake tab in the view

blourenco217 committed:
corrected serial timestamp

blourenco217 committed:
corrected distance at apply button in the servos

blourenco217 committed:
Corrected typo

blourenco217 committed:
open tab is the second

blourenco217 committed:
Corrected view of the quadature encoder

blourenco217 committed:
added teh direction input apply button

blourenco217 committed:
Enable/disable button fit the screen

blourenco217 committed:
changed label of timesatmp

blourenco217 committed:
removed duplicated advanced controls

blourenco217 committed:
corrected frame acquired active camera binding states

blourenco217 committed:
corrected labels at ports

blourenco217 committed:
corrected limits current of leds

blourenco217 committed:
Added digital input on the di/os tab

blourenco217 committed:
Cleaned up the viewmodel: unsured functions and commands

blourenco217 committed:
Added comprehensive labels on the view

blourenco217 committed:
Corrected identation at the view document

blourenco217 committed:
Corrections applied on the xmal file - improve readibility of the file
Added read me onto the the 'app' directory

blourenco217 committed:
clean up view model

blourenco217 committed:
deleted unused functionalities

blourenco217 committed:
updaated readme

blourenco217 committed:
prepare for executable of different architectures

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/harp-tech/device.behavior?pullRequest=32) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/harp-tech/device.behavior?pullRequest=32) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: blourenco217<br/>:x: Beatriz<br/><hr/>**Beatriz** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/harp-tech/device.behavior?pullRequest=32) it.</sub>

MicBoucinha reviewed (approved):
Everything seems to be in accordance with the already released applications.

From the internal testing that was made, it feels that we can now release this to the community.

MicBoucinha committed:
Add missing WriteMessagesControl

MicBoucinha committed:
Update how app version is collected

MicBoucinha committed:
Update logo for macOS

MicBoucinha committed:
Update how logos are copied to output dirs

MicBoucinha committed:
Update project to build .app on macOS

MicBoucinha committed:
Update windows installer script

MicBoucinha committed:
Change app version

MicBoucinha committed:
Fix issue with copied logos

MicBoucinha commented:
Binaries were generated and are available here: https://github.com/fchampalimaud/device.behavior/releases/tag/app1.0.0-alpha.1

glopesdev committed:
Move App project files to common folder

This is in preparation for unified releases and centralized build
infrastructure for dotnet projects.

glopesdev committed:
Update app logos and org statements

Includes all harp logo variants. As the logo works equally well in light
and dark themes, we also remove the theme variant scopes.

Ensure harp-tech org statements are consistent with other repository
components. Contributing statements are temporarily removed to make
room for an org-level CONTRIBUTING.md file.

glopesdev committed:
Rename view model source files to match class name

glopesdev committed:
Refactor for compliance with latest interface

glopesdev committed:
Remove unused package readme

glopesdev committed:
Fix nullable warnings

glopesdev reviewed (commented):
As discussed, all review comments and suggestions have been included in https://github.com/fchampalimaud/device.behavior/pull/1.

glopesdev committed:
Use description attributes for enum display

Replace the per-instance Mappings dictionary on EnumDisplayConverter
with a reflection-based lookup that reads description attributes off
enum fields, cached statically. Declare the converter and a shared item
template once at the UserControl level, and apply it to the
EnableSerialTimestamp ComboBox.

glopesdev committed:
Declare DataType on enum display template

Declaring an explicit type on the shared EnumDisplayTemplate allows the
Avalonia compiler to validate the binding under compiled bindings
without relying on inference from the surrounding ItemsControl.

glopesdev committed:
Read tooltip text from description attribute

glopesdev committed:
Use description attribute for encoder tooltips
